File: php/94-binary-tree-inorder-traversal.php
<?php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($value) { $this->val = $value; }
 * }
 */
class Solution 
{
    private $data = [];
    /**
     * @param TreeNode $root
     * @return Integer[]
     */
    public function inorderTraversal($root) 
    {
        if ($root) {
            $this->inorderTraversal($root->left);
            $this->data[] = $root->val;
            $this->inorderTraversal($root->right);
        }
        return $this->data;
    }
}

File: go/94-binary-tree-inorder-traversal.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
	var res []int
	traversal(root, &res)
	return res
}
func traversal(root *TreeNode, res *[]int) {
	if root == nil {
		return
	}
	traversal(root.Left, res)
	*res = append(*res, root.Val)
	traversal(root.Right, res)
}
